tests/test_scraper.py

```diff
@@ -12,29 +12,35 @@
     pytestmark = pytest.mark.skip(reason='Requires system to support plotting')
 
 
-def test_scraper(tmpdir, monkeypatch):
+@pytest.mark.parametrize('n_win', [1, 2])
+def test_scraper(tmpdir, monkeypatch, n_win):
     pytest.importorskip('sphinx_gallery')
     monkeypatch.setattr(pyvista, 'BUILDING_GALLERY', True)
     pyvista.close_all()
-    plotter = pyvista.Plotter(off_screen=True)
+    plotters = [pyvista.Plotter(off_screen=True) for _ in range(n_win)]
     scraper = Scraper()
     src_dir = str(tmpdir)
     out_dir = op.join(str(tmpdir), '_build', 'html')
-    img_fname = op.join(src_dir, 'auto_examples', 'images', 'sg_img.png')
+    img_fnames = [
+        op.join(src_dir, 'auto_examples', 'images', f'sg_img_{n}.png') for n in range(n_win)
+    ]
     gallery_conf = {"src_dir": src_dir, "builder_name": "html"}
     target_file = op.join(src_dir, 'auto_examples', 'sg.py')
     block = None
     block_vars = dict(
-        image_path_iterator=(img for img in [img_fname]),
+        image_path_iterator=(img for img in img_fnames),
```

Comment:
Perhaps
```suggestion
        image_path_iterator=iter(img_fnames),
```
?

Looking at this, I'm not sure why [in the scraper](https://github.com/pyvista/pyvista/blob/2096a0a4d47b33112dc04ef2184ec17b9cdf5bb1/pyvista/utilities/sphinx_gallery.py#L52-L61) we don't just do `image_path_iterator = iter(block_vars['image_paths'])`, or even better, just `zip(figures.items(), image_paths)`. Irrelevant for this PR obviously.

Reply:
Unless the idea is that multiple calls to the scraper keep using further items from the paths. In that case calling `iter()` on the input would probably still work to allow sequences to be passed in initially. Not sure.